path_testcases_test.go

```diff
@@ -0,0 +1,718 @@
+				testCases: []routeTestCase{
+					{url: "/api/v1/entity", params: []string{"entity"}, match: true},
+					{url: "/api/v1/87283827683", params: []string{"87283827683"}, match: true},
+					{url: "/api/v1/25", params: []string{"25"}, match: true},
+					{url: "/api/v1/true", params: []string{"true"}, match: true},
+				},
+			},
+			{
+				pattern: "/api/v1/:param<range(10\\,30,1500)>",
+				testCases: []routeTestCase{
+					{url: "/api/v1/entity", params: nil, match: false},
+					{url: "/api/v1/87283827683", params: nil, match: false},
+					{url: "/api/v1/25", params: []string{"25"}, match: true},
+					{url: "/api/v1/1200", params: []string{"1200"}, match: true},
+					{url: "/api/v1/true", params: nil, match: false},
+				},
+			},
+			{
+				pattern: "/api/v1/:lang<len(2)>/videos/:page<range(100,1500)>",
+				testCases: []routeTestCase{
+					{url: "/api/v1/try/videos/200", params: nil, match: false},
+					{url: "/api/v1/tr/videos/1800", params: nil, match: false},
+					{url: "/api/v1/tr/videos/100", params: []string{"tr", "100"}, match: true},
+					{url: "/api/v1/e/videos/10", params: nil, match: false},
+				},
+			},
+			{
+				pattern: "/api/v1/:lang<len(2)>/:page<range(100,1500)>",
+				testCases: []routeTestCase{
+					{url: "/api/v1/try/200", params: nil, match: false},
+					{url: "/api/v1/tr/1800", params: nil, match: false},
+					{url: "/api/v1/tr/100", params: []string{"tr", "100"}, match: true},
+					{url: "/api/v1/e/10", params: nil, match: false},
+				},
+			},
+			{
+				pattern: "/api/v1/:lang/:page<range(100,1500)>",
+				testCases: []routeTestCase{
+					{url: "/api/v1/try/200", params: []string{"try", "200"}, match: true},
+					{url: "/api/v1/tr/1800", params: nil, match: false},
+					{url: "/api/v1/tr/100", params: []string{"tr", "100"}, match: true},
+					{url: "/api/v1/e/10", params: nil, match: false},
+				},
+			},
+			{
+				pattern: "/api/v1/:lang<len(2)>/:page",
+				testCases: []routeTestCase{
+					{url: "/api/v1/try/200", params: nil, match: false},
+					{url: "/api/v1/tr/1800", params: []string{"tr", "1800"}, match: true},
+					{url: "/api/v1/tr/100", params: []string{"tr", "100"}, match: true},
+					{url: "/api/v1/e/10", params: nil, match: false},
+				},
+			},
+			{
+				pattern: "/api/v1/:date<datetime(2006\\-01\\-02)>/:regex<regex(p([a-z]+)ch)>",
+				testCases: []routeTestCase{
+					{url: "/api/v1/2005-11-01/a", params: nil, match: false},
+					{url: "/api/v1/2005-1101/paach", params: nil, match: false},
+					{url: "/api/v1/2005-11-01/peach", params: []string{"2005-11-01", "peach"}, match: true},
+				},
+			},
+			{
+				pattern: "/api/v1/:param<int>?",
+				testCases: []routeTestCase{
+					{url: "/api/v1/entity", params: nil, match: false},
+					{url: "/api/v1/8728382", params: []string{"8728382"}, match: true},
+					{url: "/api/v1/true", params: nil, match: false},
+					{url: "/api/v1/", params: []string{""}, match: true},
+				},
+			},
+		}...,
```

Comment:
Are these golang dots or spaces? 😂

Reply:
I think it's to necessary make it variadic

Reply:
golang dots for the variadic function -> for the slice combination

Reply:
I was looking at it on my phone, and got confused :-)